control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go

```diff
@@ -295,6 +309,58 @@ func ReconcileIgnitionServer(ctx context.Context,
 								EmptyDir: &corev1.EmptyDirVolumeSource{},
 							},
 						},
+						{
+							Name: "guest-kubeconfig",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									// Mount the kubeconfig for the guest cluster to fetch the FeatureGate.
+									SecretName:  hcp.Spec.KubeConfig.Name,
```

Comment:
this kubeconfig is meant for external consumers. This should be using KASServiceKubeconfigSecret at most.

Reply:
Do you know if `KASServiceKubeconfigSecret` has got permission to read the featuregate resource in the guest cluster? If not, where would I look to add that?